backend/functions/lib/adapters/gateways/case.assignment.cosmosdb.mongo.repository.test.ts

```diff
@@ -0,0 +1,117 @@
+import { CaseAssignmentCosmosMongoDbRepository } from './case.assignment.cosmosdb.mongo.repository';
+import { createMockApplicationContext } from '../../testing/testing-utilities';
+import MockData from '../../../../../common/src/cams/test-utilities/mock-data';
+import { ApplicationContext } from '../types/basic';
+import { closeDeferred } from '../../defer-close';
+import { MongoCollectionAdapter } from './mongo/mongo-adapter';
+import { getCamsError } from '../../common-errors/error-utilities';
+
+describe('offices repo', () => {
+  let context: ApplicationContext;
+  let repo: CaseAssignmentCosmosMongoDbRepository;
+
+  beforeEach(async () => {
+    context = await createMockApplicationContext();
+    repo = new CaseAssignmentCosmosMongoDbRepository(context);
+  });
+
+  afterEach(async () => {
+    await closeDeferred(context);
+    jest.restoreAllMocks();
+  });
+
+  describe('test happy paths', () => {
+    test('should create assignment', async () => {
+      const fakeAttorney = MockData.getAttorneyUser();
+      const assignment = MockData.getAttorneyAssignment({ name: fakeAttorney.name });
+      jest.spyOn(MongoCollectionAdapter.prototype, 'insertOne').mockResolvedValue(assignment.id);
+      const actual = await repo.create(assignment);
+      expect(actual).toEqual(assignment.id);
+    });
+
+    test('should update assignment', async () => {
+      const fakeAttorney = MockData.getAttorneyUser();
+      const assignment = MockData.getAttorneyAssignment({ name: fakeAttorney.name });
+      jest.spyOn(MongoCollectionAdapter.prototype, 'replaceOne').mockResolvedValue(assignment.id);
+      const actual = await repo.update(assignment);
+      expect(actual).toEqual(assignment.id);
+    });
+
+    test('should call findAssignmentsByAssignee', async () => {
+      const userId = 'userId-Joe Nobel';
+      const mockAssignments = [
+        MockData.getAttorneyAssignment({ userId }),
+        MockData.getAttorneyAssignment({ userId }),
+      ];
+      jest.spyOn(MongoCollectionAdapter.prototype, 'find').mockResolvedValue(mockAssignments);
+      const actualAssignments = await repo.findAssignmentsByAssignee(userId);
+      expect(actualAssignments).toEqual(mockAssignments);
+    });
+
+    test('should findAssignmentsByCaseId', async () => {
+      const caseId = '111-22-33333';
+      const mockAssignments = [
+        MockData.getAttorneyAssignment({ caseId }),
+        MockData.getAttorneyAssignment({ caseId }),
+      ];
+      jest.spyOn(MongoCollectionAdapter.prototype, 'find').mockResolvedValue(mockAssignments);
+      const actualAssignment = await repo.findAssignmentsByCaseId(caseId);
+
+      expect(actualAssignment).toEqual(mockAssignments);
+    });
+  });
+
+  describe('handle errors', () => {
+    const error = new Error('some error');
+
+    test('should handle error on create assignment', async () => {
```

Comment:
There are a lot of places we are using boiler plate code to test error handling. They all mock specific functions of the adapter and assert on an error. We could probably write a testing-utility function that does a lot of this for us